pkg/validate/fileobjects/raw.go

```diff
@@ -78,9 +81,19 @@ func (r *Raw) Scoped() (*Scoped, status.MultiError) {
 		DynamicNSSelectorEnabled: r.DynamicNSSelectorEnabled,
 		NSControllerState:        r.NSControllerState,
 	}
+
 	for _, obj := range r.Objects {
 		s, err := scoper.GetObjectScope(obj)
 		if err != nil {
+			// For objects with matching GVKs, all errors from scoper.GetObjectScope are skipped.
+			// This includes, but is not limited to, unknown GVK errors such as KNV1021.
+			gk := obj.GroupVersionKind().GroupKind()
+
+			if gvkutil.Matches(gk, r.SkippedGVKs) {
+				klog.V(6).Infof("ignoring KNV1021 error for %s/%s due to --no-api-server-check-for-group flag: %v", gk.Group, gk.Kind, err)
+				continue // Skip appending this error
+			}
+
 			if r.AllowUnknownKinds {
```

Comment:
nit: I'd suggest implementing the control flow as part of the `r.AllowUnknownKinds` if/else to keep this consistent. The intent is that only one of `AllowUnknownKinds` and `SkippedGVKs` is specified, and it would be more intuitive if these followed a similar control flow (rather than using a separate `if` with `continue`)